docs/performance.md

```diff
@@ -0,0 +1,254 @@
+%Cpu(s):  3.5 us,  1.7 sy,  0.0 ni, 89.9 id,  2.9 wa,  0.0 hi,  2.0 si,  0.0 st 
+MiB Mem :    970.0 total,     74.5 free,    932.7 used,     65.2 buff/cache     
+MiB Swap:      0.0 total,      0.0 free,      0.0 used.     37.3 avail Mem 
+
+    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
+   1770 root      20   0 1231828  22808      0 S   8.6   2.3   0:18.35 conformance-mys
+   1140 999       20   0 1842244 493652      0 S   4.0  49.7   0:13.93 mysqld
+```
+
+#### Steps
+
+1. Create a [GCP free tier](https://cloud.google.com/free/docs/free-cloud-features#free-tier) e2-micro VM instance in us-central1 (iowa).
+
+1. [Install Go](https://go.dev/doc/install)
+   
+   ```sh
+   instance:~$ wget https://go.dev/dl/go1.23.0.linux-amd64.tar.gz
+   instance:~$ sudo rm -rf /usr/local/go && sudo tar -C /usr/local -xzf go1.23.0.linux-amd64.tar.gz
+   instance:~$ export PATH=$PATH:/usr/local/go/bin
+   instance:~$ go version
+   go version go1.23.0 linux/amd64
+   ```
+
+1. [Install Docker using the `apt` repository](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository)
+
+   ```sh
+   https://docs.docker.com/engine/install/debian/
+   ```
+
+1. Install Git
+
+   ```sh
+   instance:~$ sudo apt-get install git -y -q
+   ...
+   instance:~$ git version
+   git version 2.39.2
+   ```
+
+1. Clone the Trillian Tessera repository
+
+   ```sh
+   instance:~$ git clone https://github.com/transparency-dev/trillian-tessera.git
+   Cloning into 'trillian-tessera'...
+   ```
+
+1. Run `cmd/conformance/mysql` and MySQL database via Docker compose
+
+   ```sh
+   instance:~/trillian-tessera$ sudo docker compose -f ./cmd/conformance/mysql/docker/compose.yaml up
+   ```
+
+1. Run `hammer` and get performance metrics
+
+   ```sh
+   hammer:~/trillian-tessera$ go run ./hammer --log_public_key=Test-Betty+df84580a+AQQASqPUZoIHcJAF5mBOryctwFdTV1E0GRY4kEAtTzwB --log_url=http://10.128.0.3:2024 --max_read_ops=0 --num_writers=512 --max_write_ops=512
+   ```
+
+### GCP Free Tier VM Instance + Cloud SQL (MySQL)
+
+**e2-micro**
+
+- vCPU: 0.25-2 vCPU (1 shared core)
+- Memory: 1 GB
+- OS: Debian GNU/Linux 12 (bookworm)
+
+**Cloud SQL (MySQL 8.0.31)**
+
+- vCPUs: 4
+- Memory: 7.5 GB
+- SSD storage: 10 GB
+
+#### Result
+
+```
+┌───────────────────────────────────────────────────────────────────────┐
+│Read (8 workers): Current max: 0/s. Oversupply in last second: 0       │
+│Write (512 workers): Current max: 2571/s. Oversupply in last second: 0 │
+│TreeSize: 2530480 (Δ 2047qps over 30s)                                 │
+│Time-in-queue: 41ms/120ms/288ms (min/avg/max)                          │
+│Observed-time-to-integrate: 568ms/636ms/782ms (min/avg/max)            │
+└───────────────────────────────────────────────────────────────────────┘
+```
+
+The bottleneck comes from the `cmd/conformance/mysql` binary on the free tier VM instance. The MySQL CPU usage is lower than 10%.
```

Comment:
Isn't this one using CloudSQL?

Reply:
Correct, this is the Cloud SQL CPU usage.

Reply:
Oops.. I forgot to capture the local machine CPU usage.

Reply:
I will add the missing data in the upcoming pull request.